pages/api/account/update-pin/[id].ts

```diff
@@ -0,0 +1,61 @@
+import type { NextApiResponse, NextApiRequest } from "next"
+import sierraClient from "../../../../src/server/sierraClient"
+import initializePatronTokenAuth from "../../../../src/server/auth"
+
+/**
+ * API route handler for /api/account/update-pin/{patronId}
+ */
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  let responseMessage
+  let responseStatus
+  const patronTokenResponse = await initializePatronTokenAuth(req.cookies)
+  const cookiePatronId = patronTokenResponse.decodedPatron?.sub
+  if (!cookiePatronId) {
+    responseStatus = 403
+    responseMessage = "No authenticated patron"
+    return res.status(responseStatus).json(responseMessage)
+  }
+  if (req.method == "PUT") {
+    /**  We get the patron id from the request: */
+    const patronId = req.query.id as string
+    const oldPin = req.body.old
+    const newPin = req.body.new
+    const barcode = req.body.barcode
+    /**  We check that the patron cookie matches the patron id in the request,
+     * i.e.,the logged in user is updating their own PIN. */
+    if (patronId == cookiePatronId) {
+      const response = await pinUpdate(patronId, barcode, oldPin, newPin)
+      responseStatus = response.status
+      responseMessage = response.message
+    } else {
+      responseStatus = 403
+      responseMessage = "Authenticated patron does not match request"
+    }
+  }
+  res.status(responseStatus).json(responseMessage)
+}
+
+export async function pinUpdate(
+  patronId: string,
+  barcode: string,
+  oldPin: string,
+  newPin: string
+) {
+  try {
+    const client = await sierraClient()
+    await client.post("patrons/validate", {
+      barcode: barcode,
+      pin: oldPin,
+    })
+    await client.put(`patrons/${patronId}`, { pin: newPin })
```

Comment:
This is the similar line that `settingsUpdate` calls. I wonder if it's possible to somehow call `settingsUpdate` here but that might be more complicated and not worth doing.

Just pointing this out because now there are two places where the same Sierra endpoint is called. And really, this internal API route only want to verify the old pin before updating the patron's setting. Something we can discuss at some point with Vera but this work so let's proceed.

Reply:
Right. It does seem like we could do something along the lines of, if the pin is part of the request body, then do this validation, otherwise just update the settings (all in `settingsUpdate`). I think they were separated because they're separated in the UI ("Save changes" for settings vs. "Change PIN/password" buttons)? There's already quite a lot of repetition with these handlers though so I don't know if it's worth it to just refactor this one part

Reply:
for this pr let's just combine these two endpoints into one handler. which i believe means they have to share a url, which is fine.
we can investigate combining everything later.